# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieCommons.h
# Commit: cc3c39aa8fb8
# Full Hash: cc3c39aa8fb8124cca265e84f71a29cbd218b704
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 2fc9a3d0ad8b -r cc3c39aa8fb8 netwerk/cookie/CookieCommons.h
--- a/netwerk/cookie/CookieCommons.h	Wed Sep 11 16:26:54 2024 +0000
+++ b/netwerk/cookie/CookieCommons.h	Wed Sep 11 16:26:55 2024 +0000
@@ -69,6 +69,9 @@
 
   static bool PathMatches(Cookie* aCookie, const nsACString& aPath);
 
+  static bool PathMatches(const nsACString& aCookiePath,
+                          const nsACString& aPath);
+
   static nsresult GetBaseDomain(nsIEffectiveTLDService* aTLDService,
                                 nsIURI* aHostURI, nsACString& aBaseDomain,
                                 bool& aRequireHostMatch);
@@ -145,6 +148,26 @@
 
   static void GetServerDateHeader(nsIChannel* aChannel,
                                   nsACString& aServerDateHeader);
+
+  enum class SecurityChecksResult {
+    // A sandboxed context detected.
+    eSandboxedError,
+    // A security error needs to be thrown.
+    eSecurityError,
+    // This context should not see cookies without returning errors.
+    eDoNotContinue,
+    // No security issues found. Proceed to expose cookies.
+    eContinue,
+  };
+
+  // Runs the security checks requied by specs on the current context (Document
+  // or Worker) to see if it's allowed to set/get cookies. In case it does
+  // (eContinue), the cookie principals are returned. Use the
+  // `aCookiePartitionedPrincipal` to retrieve CHIP cookies. Use
+  // `aCookiePrincipal` to retrieve non-CHIP cookies.
+  static SecurityChecksResult CheckGlobalAndRetrieveCookiePrincipals(
+      mozilla::dom::Document* aDocument, nsIPrincipal** aCookiePrincipal,
+      nsIPrincipal** aCookiePartitionedPrincipal);
 };
 
 }  // namespace net